storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -286,34 +288,32 @@ public void doRegistration(final SchedulerDriver driver, Protos.FrameworkID id)
     _state.put(FRAMEWORK_ID, id.getValue());
     _offers = new HashMap<Protos.OfferID, Protos.Offer>();
 
-    if (_enabledLogviewerSidecar) {
-
-      _timer.scheduleAtFixedRate(new TimerTask() {
-        @Override
-        public void run() {
-          // performing "explicit" reconciliation; master will respond with the latest state for all logviewer tasks
-          // in the framework scheduler's statusUpdate() method
-          List<TaskStatus> taskStatuses = new ArrayList<TaskStatus>();
-          List<String> logviewerPaths = _zkClient.getChildren(_logviewerZkDir);
-          if (logviewerPaths == null) {
-            _driver.reconcileTasks(taskStatuses);
-            return;
-          }
-          for (String path : logviewerPaths) {
-            TaskID logviewerTaskId = TaskID.newBuilder()
-                                           .setValue(new String(_zkClient.getNodeData(String.format("%s/%s", _logviewerZkDir, path))))
-                                           .build();
-            TaskStatus logviewerTaskStatus = TaskStatus.newBuilder()
-                                                       .setTaskId(logviewerTaskId)
-                                                       .setState(TaskState.TASK_RUNNING)
-                                                       .build();
-            taskStatuses.add(logviewerTaskStatus);
-          }
+    _timer.scheduleAtFixedRate(new TimerTask() {
+      @Override
+      public void run() {
+        // performing "explicit" reconciliation; master will respond with the latest state for all logviewer tasks
+        // in the framework scheduler's statusUpdate() method
+        List<TaskStatus> taskStatuses = new ArrayList<TaskStatus>();
+        List<String> logviewerPaths = _zkClient.getChildren(_logviewerZkDir);
```

Comment:
Is this the line of code that is generating the log line that you mention as the visible side-effect of this change?

Reply:
Correct. The more I think about this (and having seen the code run on our systems), I think this trace will be a nuisance, given how frequently the timer task runs (every 5 mins). I'm in favor of adding an explicit check to first test that the `logviewer` ZNode exists (without the error trace) before trying to get its children. Thoughts?

Reply:
Sorry for the incredibly slow response. I like this solution, please implement.